### PR TITLE
Feature/pagination

### PR DESCRIPTION
**Pull request by ira-alchynska**
add function pagination

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -13,7 +13,8 @@
     <main>
       <div id="main-page"></div>
     </main>
-
+    <div id="exercises-container" class="exercises-container"></div>
+    <load src="./partials/pagination.html" />
```
Comment (snegurova) [inline src/index.html]: this should go to page/home.html

File: src/main.js [hunk 2]
```diff
@@ -9,6 +10,7 @@ links.forEach((link) => link.addEventListener('click', route));
 handleLocation();
 
 updateQuote();
+import './js/pagination';
```
Comment (snegurova) [inline src/main.js]: All imports go first, then code

File: src/pages/exercises.html [hunk 3]
```diff
@@ -1,5 +1,8 @@
 <div class="container">
   <h1>Exercises</h1>
 
+ 
+  </div>
```
Comment (snegurova) [inline src/pages/exercises.html]: Please remove

File: src/index.html (hunk 1 above)
Comment (ira-alchynska) [reply src/index.html]: done

File: src/main.js (hunk 2 above)
Comment (ira-alchynska) [reply src/main.js]: done

File: src/pages/exercises.html (hunk 3 above)
Comment (ira-alchynska) [reply src/pages/exercises.html]: done

File: src/js/pagination.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    pageInfo.innerHTML = pages
+      .map(page => {
+        if (page === "...") {
+          return `<span class="dots">${page}</span>`;
+        }
+        return `<button class="page-number ${page === currentPage ? 'active' : ''}" data-page="${page}">${page}</button>`;
+      })
+      .join(" ");
+  
+    callback(currentPage);
```
Comment (snegurova) [inline src/js/pagination.js]: the callback should be added as event listener to each button, button should have value attribute with page number, so it's reachable in the callback in event, and you don't need to call callback on each render, it should be called on button click

To test if it works, create callback, that will log to console number of the button (event.target.value)

File: src/js/pagination.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    }
+  }
+
+  async function fetchAndRenderExercises(page) {
+    const data = await getExercises(page);
+    console.log("Current Page Data:", data.results);
+
+  }
+
+  firstBtn.addEventListener("click", () => goToPage(1));
```
Comment (snegurova) [inline src/js/pagination.js]: here the same callback should be added as event listener on click, the button should have propriet number in the value attribute

File: src/js/api/ExercisesService.js [hunk 6]
```diff
@@ -7,6 +7,7 @@ export default class ExercisesService {
   #page = 1;
   #filter = '';
   #name = '';
+
```
Comment (snegurova) [inline src/js/api/ExercisesService.js:10]: Try to avoid only formating changes, it leads to additional merge conflicts

File: src/index.html [hunk 7]
```diff
@@ -13,11 +13,12 @@
...
       <div id="main-page"></div>
     </main>
-
+    <div id="exercises-container" class="exercises-container"></div>
+   
     <load src="./partials/footer.html" />
 
     <load src="partials/modal.html" />
-
+    <load src="./partials/pagination.html"/>
```
Comment (snegurova) [inline src/index.html:21]: Please remove pagination from index

File: src/js/pagination.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
+document.addEventListener("DOMContentLoaded", async () => {
+  setupPagination(fetchAndRenderExercises);
+});
```
Comment (snegurova) [inline src/js/pagination.js]: move it after you have declared function fetchAndRenderExercises

File: src/js/pagination.js [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+export const getContentPagination = () => {
+  firstBtn = document.getElementById("first-btn");
+  prevBtn = document.getElementById("prev-btn");
+  nextBtn = document.getElementById("next-btn");
+  lastBtn = document.getElementById("last-btn");
+  pageInfo = document.getElementById("page-info");
+};
+
+async function setupPagination(callback) {
+  getContentPagination();
```
Comment (snegurova) [inline src/js/pagination.js:30]: you don't need to call it here, it is called in handleLocation function in routing

File: src/js/pagination.js [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+
+  const initialData = await getExercises(1);
+  const totalPages = initialData.totalPages;
```
Comment (snegurova) [inline src/js/pagination.js]: For now, let's leave it here, the number of pages should be a parameter for pagination function

File: src/js/pagination.js [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+  let currentPage = 1;
+
+  function renderPagination() {
```
Comment (snegurova) [inline src/js/pagination.js]: Please move this function out of setupPagination and pass needed params, in the setupPagination only call when needed

File: src/js/pagination.js [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+    
+ 
+    document.querySelectorAll(".page-number").forEach(button => {
+      button.addEventListener("click", (event) => {
+        goToPage(Number(event.target.value), callback);
+      });
+    });
+  }
+  
+  async function goToPage(page, callback) {
```
Comment (snegurova) [inline src/js/pagination.js]: Please move this function out of setupPagination and pass needed params, in the setupPagination only call when needed. The function is not async, we do not wait for callback call

File: src/js/pagination.js [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+
+  firstBtn.addEventListener("click", () => goToPage(1, callback));
+  prevBtn.addEventListener("click", () => goToPage(currentPage - 1, callback));
+  nextBtn.addEventListener("click", () => goToPage(currentPage + 1, callback));
+  lastBtn.addEventListener("click", () => goToPage(totalPages, callback));
+
+ 
+  await callback(currentPage);
```
Comment (snegurova) [inline src/js/pagination.js]: this call is not needed, we call callback only on click in goToPage function

File: src/js/pagination.js (hunk 8 above)
Comment (ira-alchynska) [reply src/js/pagination.js]: done

File: src/js/pagination.js (hunk 9 above)
Comment (ira-alchynska) [reply src/js/pagination.js:30]: it does not work let's check in a call together

File: src/index.html (hunk 7 above)
Comment (ira-alchynska) [reply src/index.html:21]: is is not working in home still lets check together

File: src/js/pagination.js (hunk 11 above)
Comment (ira-alchynska) [reply src/js/pagination.js]: done

File: src/js/pagination.js (hunk 12 above)
Comment (ira-alchynska) [reply src/js/pagination.js]: moved

File: src/js/pagination.js (hunk 13 above)
Comment (ira-alchynska) [reply src/js/pagination.js]: deleted

**Review by snegurova — APPROVED** [review]

**Pull request merged**